### PR TITLE
feat: Onboard FDIC dataset

### PR DESCRIPTION
**Pull request by gkodukula**
## Description

Dataset: FDIC

- Locations
- Institutions

## Checklist

- [x] **(Required)** This pull request is appropriately labeled
- [x] Please merge this pull request after it's approved

Use the sections below based on what's applicable to your PR and delete the rest:

### Feature
- [x] I'm adding or editing a feature
- [x] I have updated the [`README`](https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/main/README.md) accordingly
- [x] I have added/revised tests for the feature

### Data Onboarding
- [x] I'm adding or editing a dataset
- [x] The [Google Cloud Datasets team](mailto:cloud-datasets-onboarding@google.com) is aware of the proposed dataset
- [x] I put all my code inside  `datasets/fdic` and nothing outside of that directory

### CODE REVIEW

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    headers: typing.List[str],
+    rename_mappings: dict,
+    pipeline_name: str,
+) -> None:
+
+    logging.info(
+        f"FDIC{pipeline_name} process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:43]: Remove all blank lines within functions

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            "Int64", errors="ignore"
+        )
+        logging.info("Replacing date values...")
+        format_date_list = ["date_established", "last_updated"]
+        format_date(format_date_list, df)
+        logging.info("Replacing with null values...")
+        df["cbsa_metro_fips_code"] = df["cbsa_metro_fips_code"].replace(0, "NULL")
+    else:
+        logging.info("Replacing bool values...")
+        replace_bool_list = [
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:78]: Please move to pipeline.yaml and incorporate as a variable

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            "ownership_type",
+            "csa_indicator",
+            "cbsa_metro_flag",
+            "cbsa_micro_flag",
+            "cbsa_division_flag",
+            "subchap_s_indicator",
+        ]
+        replace_bool(replace_bool_list, df)
+        logging.info("Replacing date values...")
+        replace_date_list = [
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:109]: Please move to pipeline.yaml and incorporate as a variable

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+            "last_structural_change",
+            "report_date",
+            "reporting_period_end_date",
+            "run_date",
+            "cfpb_supervisory_start_date",
+            "cfpb_supervisory_end_date",
+        ]
+        replace_date(replace_date_list, df)
+        logging.info("Formatting date values...")
+        format_date_list = [
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:123]: Please move to pipeline.yaml and incorporate as a variable

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            "end_effective_date",
+            "deposit_insurance_date",
+            "last_structural_change",
+            "report_date",
+            "reporting_period_end_date",
+            "run_date",
+        ]
+        format_date(format_date_list, df)
+        logging.info("Filling null values...")
+        null_list = [
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:136]: Please move to pipeline.yaml and incorporate as a variable

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+    main(
+        source_url=os.environ["SOURCE_URL"],
+        source_file=pathlib.Path(os.environ["SOURCE_FILE"]).expanduser(),
+        target_file=pathlib.Path(os.environ["TARGET_FILE"]).expanduser(),
+        target_gcs_bucket=os.environ["TARGET_GCS_BUCKET"],
+        target_gcs_path=os.environ["TARGET_GCS_PATH"],
+        headers=json.loads(os.environ["CSV_HEADERS"]),
+        rename_mappings=json.loads(os.environ["RENAME_MAPPINGS"]),
+        pipeline_name=os.environ["PIPELINE_NAME"],
```
Comment (nlarge-google) [inline datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:239]: Add respective variables here (see previous comments)

**Review by nlarge-google — CHANGES_REQUESTED** [review]
Great Job.  Just need to make the respective changes and we are good to go.

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 6 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py:164]: done

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 5 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py]: done

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 4 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py]: done

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 3 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py]: done

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 2 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py]: done

File: datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 1 above)
Comment (gkodukula) [reply datasets/fdic/pipelines/_images/run_csv_transform_kub/csv_transform.py]: done

**Review by nlarge-google — APPROVED** [review]
Great Job!

**Comment by nlarge-google** [comment]
Merged and deployed to production

**Pull request merged**